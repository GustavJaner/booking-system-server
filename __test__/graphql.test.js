const { graphql } = require('graphql');
const {MongoClient} = require('mongodb');

const schema = require('../graphql/schema');
const post = require('../models/post/post');
const credentials = require('../credentials');

var rp = require('request-promise');
const API = 'http://localhost:5000/graphql'

describe('Testing Database + GraphQL', () => {
  let connection;
  let db;

  beforeAll(async () => {
    connection = await MongoClient.connect(credentials.MONGO_URI, { 
        useUnifiedTopology: true,
        useNewUrlParser: true
     });
    db = await connection.db('graphql');

  });

  afterAll(async () => {
    await connection.close();
    await db.close();
  });

  
  test('Query with GraphQL', async () => {
    //language=GraphQL
    const query = `
        query{
            posts {    
                id
                title
                content
            }
        }
    `;

    const rootValue = {};
    const context = () => ({id, title, content})
  
    const response = await rp({method: 'POST', uri: API, body: {query}, json: true});
    expect(response).toMatchSnapshot();

    //const result = await graphql(schema, query, rootValue, context);
    //console.log(result)
  });
});
