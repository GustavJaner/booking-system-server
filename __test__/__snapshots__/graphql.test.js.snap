// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Database + GraphQL Query with GraphQL 1`] = `
Object {
  "data": Object {
    "posts": Array [
      Object {
        "content": "test",
        "id": "5ddbe5847ceb39a3be31146b",
        "title": "string",
      },
      Object {
        "content": "mmm",
        "id": "5ddbf6a5f9fbc0b250dd71b7",
        "title": "mumma",
      },
      Object {
        "content": "mmmasdasd",
        "id": "5ddbf6b7f9fbc0b250dd71b8",
        "title": "jajajfeef",
      },
    ],
  },
}
`;
